src/main/java/org/opentripplanner/middleware/controllers/api/OtpUserController.java

```diff
@@ -88,23 +91,35 @@ protected OtpUser getUserProfile(Auth0UserProfile profile) {
      * HTTP endpoint to send an SMS text to an {@link OtpUser}'s phone number with a verification code. This is used
      * during user signup (or if a user wishes to change their notification preferences to use a new un-verified phone
      * number).
+     * Before sending a SMS request, this endpoint also saves the submitted phone number, marks it as not verified.
      */
     public VerificationResult sendVerificationText(Request req, Response res) {
         OtpUser otpUser = getEntityForId(req, res);
-        if (otpUser.phoneNumber == null) {
-            logMessageAndHalt(req, HttpStatus.NOT_FOUND_404, "User must have valid phone number to verify.");
+        // Get phone number from the path param.
+        String phoneNumber = req.params(PHONE_PARAM);
+        if (StringUtils.isBlank(phoneNumber)) {
+            logMessageAndHalt(req, HttpStatus.BAD_REQUEST_400, "A phone number must be provided.");
         }
-        Verification verification = NotificationUtils.sendVerificationText(otpUser.phoneNumber);
+
+        // Update OtpUser before submitting SMS request.
+        otpUser.phoneNumber = phoneNumber;
```

Comment:
This approach overwrites the existing phone number before it is verified. If we do things this way, it will be impossible to send off alerts to the phone until it is verified. And if the user decides they actually don't want to verify after all and keep their old number, we will have overwritten it and they would have to go back in and reconfirm their existing phone number.

The temporary phone number should be stored in a temporary field until it is verified.

Reply:
See the Revert Number button in https://github.com/opentripplanner/otp-react-redux/pull/224 for part of this concern.

Reply:
I disagree with the approach of having a revert option due to some of the concerns shared above. Additionally, it is more code to maintain than keeping an existing phone number intact until verification and requiring the user to reverify an original number if they want to change it back.